nupic/algorithms/anomaly.py

```diff
@@ -160,6 +160,39 @@ def compute(self, activeColumns, predictedColumns,
     return score
 
 
+  @staticmethod
+  def initFactory(slidingWindowSize, mode, binaryAnomalyThreshold):
+    """
+    factory method for initiation of Anomaly() instance using parameters as text strings, 
+    useful eg for OPF. 
+    This method parses the parameters and calls the default __init__() properly. 
+    For details of the parameters see __init__. 
+    Use None to automatically supply default values.
+
+    @param slidingWindowSize (str) length of sliding window
+    @param mode (str) Anomaly mode
+    @param binaryAnomalyThreshold (str) binary discretization threshold
+    @return Anomaly instance, call Anomaly.initFactory(....) 
+    """
+    # defaults
+    if slidingWindowSize is None:
+      slidingWindowSize = "0"
+    if mode is None:
+      mode = "pure"
+    if binaryAnomalyThreshold is None:
+      binaryAnomalyThreshold = "0.0"
+
+    try:
+      iWindow = int(slidingWindowSize)
+      fThreshold = float(binaryAnomalyThreshold)
```

Comment:
Ok, I see what you're doing here, but I still think they should be initialized in the correct type above.